lib/safer_rails_console/patches/sandbox/transaction_read_only.rb

```diff
@@ -11,13 +11,27 @@ def begin_db_transaction
           end
         end
 
+        module MySQLPatch
+          def begin_db_transaction
+            execute 'SET TRANSACTION READ ONLY'
+            super
+          end
+        end
+
         if defined?(::ActiveRecord::ConnectionAdapters::PostgreSQLAdapter)
           ::ActiveRecord::ConnectionAdapters::PostgreSQLAdapter.prepend(PostgreSQLAdapterPatch)
 
           # Ensure transaction is read-only if it was began before this patch was loaded
           connection = ::ActiveRecord::Base.connection
           connection.execute 'SET TRANSACTION READ ONLY' if connection.open_transactions > 0
         end
+
+        if defined?(::ActiveRecord::ConnectionAdapters::AbstractMysqlAdapter)
+          ::ActiveRecord::ConnectionAdapters::AbstractMysqlAdapter.prepend(MySQLPatch)
+
+          # Not possible to change a running transaction to read-only in MySQL
+          # https://dev.mysql.com/doc/refman/8.4/en/set-transaction.html
```

Comment:
I don't think so. I'm having a hard time thinking of scenarios where this would even be possible.
Not sure if this is different in Postgres but in Mysql, since this is a new session, there shouldn't be any transactions running for this session.

There could, however, be running transactions for other session that, even if we could, we wouldn't want to make read-only. For this reason, throwing an error or informing the user doesn't make much sense to me.

I may be missing some scenarios though.

Reply:
I'm not to sure either. Unfortunately, I wasn't able to glean anything from PR that originally added it for the Postgres version.